### PR TITLE
chore: Update to Twitch4J 1.14.0

### PR DESCRIPTION
**Pull request by Awakened-Redstone**
### Prerequisites for Code Changes

* [x] This pull request follows the code style of the project
* [x] I have tested this feature

### Changes Proposed

* Update to Twitch4J 1.14.0
* More strict publishing
* Auto fetch Minecraft versions for Modrinth publishing
* New version scheme (`twitch4j_version+mod_version
* Added support for a `CHANGELOG.md` file for a more complete changelog when publishing to Modrinth
* Removed sources jar as it has no use as of now
* Updated Loom to `1.1-SNAPSHOT`
* Updated loader version to `0.14.7`

### Additional Information
New version scheme rules:
* The mod version should bump by a minor every Twitch4J version
* The mod version patch should only increase if there is a small change to the mod itself, this includes changes to the gradle that affects the final jar
* The mod minor should only increase (outside of Twitch4J version bump) if there is a significant, but not breaking, change to the mod that may affect mods that use it
* The mod version major should only increase if there is a breaking change for mods that use it, such as mod id change

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -154,23 +154,72 @@ modrinth {
...
+        def i = 0
+        for (def version : versions) {
+            if (version.stable) {
+                if (i > 0) {
+                    output.append("\n")
+                }
+                output.append("${version.version}")
+                i++
+            }
+        }
```
Comment (iProdigy) [inline build.gradle]: Would this work or would `readLines()` include an empty string at the end?
```suggestion
        for (def version : versions) {
            if (version.stable) {
                output.append("${version.version}")
                output.append("\n")
            }
        }
```

File: build.gradle [hunk 2]
```diff
@@ -154,23 +154,72 @@ modrinth {
...
-        List<String> projectRC = projectVersionNumber[1].split("[.]", 2)
-        projectVersionName = "${projectVersionNumber[0]} - Release Candidate ${projectRC[1]}"
-        versionType = "beta"
     }
 
     token = System.getenv("MODRINTH_TOKEN")
     projectId = "CmHD69Pj"
     versionName = projectVersionName
-    changelog = "The changelog can be seen at https://github.com/twitch4j/twitch4j/releases/tag/v${project.version}"
+    changelog = "### The Twitch4J changelog can be seen at https://github.com/twitch4j/twitch4j/releases/tag/v${project.version}\n\n" + CHANGELOG
```
Comment (iProdigy) [inline build.gradle]:
```suggestion
    changelog = "### The Twitch4J changelog can be seen at https://github.com/twitch4j/twitch4j/releases/tag/v${project.library_version}\n\n" + CHANGELOG
```

File: build.gradle [hunk 3]
```diff
@@ -154,23 +154,72 @@ modrinth {
...
-    if (projectVersion.contains("beta")) {
-        List<String> projectBeta = projectVersionNumber[1].split("[.]", 2)
+    if (projectVersion.contains("patch")) {
+        List<String> _projectVersionNumber = projectVersion.split("\\+", 2)
+        List<String> projectPatch = _projectVersionNumber[1].split("\\.", 2)
+        projectVersionName = "${_projectVersionNumber[0]} - Patch ${projectPatch[1]}"
+    } else if (projectVersion.contains("rc")) {
+        List<String> projectRC = projectVersionNumber[1].split("\\.", 2)
+        projectVersionName = "${projectVersionNumber[0]} - Release Candidate ${projectRC[1]}"
+        versionType = "beta"
```
Comment (iProdigy) [inline build.gradle]: c+v typo?

```suggestion
        versionType = "rc"
```

File: build.gradle [hunk 4]
```diff
@@ -154,23 +154,72 @@ modrinth {
     String projectVersion = project.version.toString()
     List<String> projectVersionNumber = projectVersion.split("-", 2)
     String projectVersionName = "Release ${projectVersionNumber[0]}"
-    if (projectVersion.contains("beta")) {
-        List<String> projectBeta = projectVersionNumber[1].split("[.]", 2)
+    if (projectVersion.contains("patch")) {
+        List<String> _projectVersionNumber = projectVersion.split("\\+", 2)
+        List<String> projectPatch = _projectVersionNumber[1].split("\\.", 2)
```
Comment (iProdigy) [inline build.gradle]: since this is used in every branch, could it be defined once (before the if statement)

File: .gitignore [hunk 5]
```diff
@@ -116,3 +116,4 @@ run/
 
 # Avoid ignoring Gradle wrapper jar file (.jar files are usually ignored)
 !gradle-wrapper.jar
+/CHANGELOG.md
```
Comment (iProdigy) [inline .gitignore:119]: if this is a custom changelog pertaining to the mod (not t4j), i wouldn't mind it being tracked in the repo, but up to you

**Review by iProdigy — COMMENTED** [review]
Nice work

File: build.gradle (hunk 3 above)
Comment (Awakened-Redstone) [reply build.gradle]: Modrinth doesn't have an RC version type, so I use beta instead

File: build.gradle (hunk 1 above)
Comment (Awakened-Redstone) [reply build.gradle]: I imagine so, this is a code I extracted from Fabric Language Kotlin

Comment (Awakened-Redstone) [reply build.gradle]: It does not

Comment (Awakened-Redstone) [reply build.gradle]: Change included on [chore: Improve build.gradle](https://github.com/twitch4j/twitch4j-fabric/pull/1/commits/8fe0b0d5f88998af1c9b8f26f2cd1c533c741d67)

File: .gitignore (hunk 5 above)
Comment (Awakened-Redstone) [reply .gitignore:119]: I don't really see a reason to keep it there as it is just a file to make it easier to write the changelog for Modrinth

File: build.gradle (hunk 4 above)
Comment (Awakened-Redstone) [reply build.gradle]: Modified it to do so and likely improve the system a bit

**Review by iProdigy — APPROVED** [review]
LGTM!

**Pull request merged**